### PR TITLE
 Azure RedHat Openshift (ARO) support

### PR DESCRIPTION
**Pull request by jbianquetti-nami**
Add support for ARO in cloud custodian 

Example policy:

```
policies:
  - name: broken-open-shift
    resource: azure.open-shift
    filters:
      - type: value
        key: name
        op: equal
        value_type: normalize
        value: environment-64ozd
    actions:
      - type: delete
```

### CODE REVIEW

**Comment by jbianquetti-nami** [comment]
@kapilt Would you take a look at this? Thanks.

File: tools/c7n_azure/c7n_azure/resources/aro.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+from c7n_azure.provider import resources
+from c7n_azure.resources.arm import ArmResourceManager
+
+
+@resources.register('aro')
```
Comment (kapilt) [inline tools/c7n_azure/c7n_azure/resources/aro.py:8]: just thinking for simplicity on name how about just `open-shift` ?

**Comment by kapilt** [comment]
ci seems to be failing on docs due to `/home/runner/work/cloud-custodian/cloud-custodian/docs/source/azure/resources/aro.rst: WARNING: document isn't included in any toctree
`

I do wonder if thats an artifact of caching in the docs ci independent of this pr.

**Review by kapilt — APPROVED** [review]
thanks, lgtm

**Pull request merged**